src/App.tsx

```diff
@@ -15,71 +15,93 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+interface S {
+  selectedGood: string;
+}
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+// export const App: React.FC = () => (
+export class App extends React.Component<{}, S> {
+  state = {
+    selectedGood: 'Jam',
+  };
+  // React.MouseEvent<HTMLButtonElement>
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  handleGoodButton = (event: React.MouseEvent<HTMLButtonElement>,
+    good: string) => {
+    const button = event.currentTarget;
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+    if (button.dataset.cy === 'AddButton') {
+      button.dataset.cy = 'RemoveButton';
+      this.setState({ selectedGood: good });
+    } else {
+      button.dataset.cy = 'AddButton';
+      this.setState({ selectedGood: '' });
+    }
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  handleClearButton = () => {
+    this.setState({ selectedGood: '' });
+  };
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+  render() {
+    const { selectedGood } = this.state;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+    return (
+      <main className="section container">
+        <h1 className="title is-flex is-align-items-center">
+          {`${selectedGood ? `${selectedGood} is` : 'No goods'} selected` }
+          {selectedGood
+            ? (
+              <>
+                {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+                <button
+                  data-cy="ClearButton"
+                  type="button"
+                  className="delete ml-3"
+                  onClick={this.handleClearButton}
+                />
+              </>
+            )
+            : null}
```

Comment:
Don't do that, you have `&&` operator
Also, remove extra fragment
```suggestion
          {selectedGood && (
                {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
                <button
                  data-cy="ClearButton"
                  type="button"
                  className="delete ml-3"
                  onClick={this.handleClearButton}
                />
            )}
```

Reply:
in next commit used &&
but if delete the fragment it will not work, also tried to add wrapper({} or ()) and it didn't help so I keep fragment